nupic/algorithms/anomaly.py

```diff
@@ -22,34 +22,43 @@
 """Anomaly-related algorithms."""
 
 import numpy
+import collections
 
 from nupic.algorithms.anomaly_likelihood import AnomalyLikelihood
 from nupic.utils import MovingAverage
 
 
 def computeRawAnomalyScore(activeColumns, prevPredictedColumns):
-  """Computes the raw anomaly score.
+    """Computes the raw anomaly score.
 
-  The raw anomaly score is the fraction of active columns not predicted.
+    The raw anomaly score is the fraction of active columns not predicted.
 
-  @param activeColumns: array of active column indices
-  @param prevPredictedColumns: array of columns indices predicted in prev step
-  @return anomaly score 0..1 (float)
-  """
-  nActiveColumns = len(activeColumns)
-  if nActiveColumns > 0:
-    # Test whether each element of a 1-D array is also present in a second
-    # array. Sum to get the total # of columns that are active and were
-    # predicted.
-    score = numpy.in1d(activeColumns, prevPredictedColumns).sum()
-    # Get the percent of active columns that were NOT predicted, that is
-    # our anomaly score.
-    score = (nActiveColumns - score) / float(nActiveColumns)
-  else:
-    # There are no active columns.
-    score = 0.0
+    The implementation is using in1D() function, 
+    "satisfied" means it only cares if activeColumns at time T have been 
+    predicted at T-1. 
+
+    Which means a temporal pooler TP with all columns in predictive state at T-1
```

Comment:
This reads a little funny. I'm not sure if the intent is to help users understand the computation better or point out this particular use case. Perhaps this wording?

"If all columns were predicted at previous time step (prevPredictedColumns) then the score will be zero regardless of which columns, if any, are active currently (activeColumns)."

Specifically this removes the reference to "temporal pooler TP" and the notion of T and T-1, which a programmer may not be familiar with.